packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeUtils.ts

```diff
@@ -70,100 +69,33 @@ export class EscapedPath {
 	public static create(path: string): EscapedPath {
 		return new EscapedPath(encodeURIComponent(path));
 	}
-	public static createAndConcat(pathParts: string[]): EscapedPath {
-		let ret = EscapedPath.create(pathParts[0] ?? "");
-		for (let i = 1; i < pathParts.length; i++) {
-			// Non null asserting here since we are iterating over pathParts
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			ret = ret.concat(EscapedPath.create(pathParts[i]!));
-		}
-		return ret;
-	}
 	public toString(): string {
 		return this.path;
 	}
+	public concatWith(path: EscapedPath, concatWith: string): EscapedPath {
```

Comment:
It's not obvious what this function does based on the param names.
I would rename the params to something like `path1: string` and `path2: EscapedPath` so it's clear what order they will be concatenated in.
Also, maybe you could just accept strings and do the `EscapedPath.create` part here to keep it consistent.

Reply:
updated funciton name and signature